scala/friesian/src/main/scala/com/intel/analytics/bigdl/friesian/nearline/recall/RecallNearlineUtils.scala

```diff
@@ -43,12 +45,22 @@ object RecallNearlineUtils {
       var df = spark.read.parquet(parquetFiles: _*)
       df = df.select(itemFeatureColumns.map(col): _*).distinct()
       val data = df.rdd.map(row => {
-        val id = row.getInt(0)
-        val data = row.getAs[DenseVector](1).toArray.map(_.toFloat)
+        val id = row.get(0)
+        val data = row.get(1)
         (id, data)
       }).collect()
-      val resultFlattenArr = data.flatMap(_._2)
-      val idList = data.map(_._1)
+
+      val resultFlattenArr = if (data.length > 0) {
+        val tmpDataArr = data.map(_._2)
+        (if (tmpDataArr(0).isInstanceOf[DenseVector]) {
+          tmpDataArr.flatMap(_.asInstanceOf[DenseVector].toArray.map(_.toFloat))
+        } else {
```

Comment:
Please raise an exception if tmpDataArr(0) is neither DenseVector nor WrappedArray.

Reply:
You can use match case here
```
tmpDataArr(0) match {
case _: DenseVector => ....
.....
case d =>
 Log4Error.....
```